lib/Doctrine/DBAL/Driver/PDOSqlsrv/Connection.php

```diff
@@ -40,6 +40,22 @@ public function __construct($dsn, $user = null, $password = null, array $options
     /**
      * {@inheritDoc}
      */
+    public function lastInsertId($name = null)
+    {
+        if (null === $name) {
+            return parent::lastInsertId($name);
+        }
+
+        $sql = 'SELECT CONVERT(VARCHAR(MAX), current_value) FROM sys.sequences WHERE name = ?';
+        $stmt = $this->prepare($sql);
+        $stmt->execute(array($name));
+
+        return $stmt->fetchColumn();
```

Comment:
Given to [the documentation](https://msdn.microsoft.com/en-us/en-en/library/ff877934.aspx) no.